news/_posts/2019-12-25-prestashop-in-2019-and-beyond-part-3-the-future-architecture.md

```diff
@@ -0,0 +1,269 @@
+
+Similarly, since last year we have been making architecture choices that may have seemed arbitrary or incomprehensible to some. "Why are you making it so needlessly complicated?", they said. Rest assured, **there is a good reason**. Each choice we make answers to a specific problem in the context of a bigger long-term vision. We just needed time to put this reasoning into words (or should I say, articles), so that everyone understands not just _what_ we are planning to do, but more importantly, _why_ we are leading the project in this direction.
+
+## What drives a new architecture?
+
+When we began thinking about a new architecture, we started with a simple exercise: if we had to build PrestaShop again, what would its architecture look like?
+
+### A solution for every problem
+
+Our first job was trying to understand what our current problems were and where they were coming from–this is what the previous articles were about. Once we knew what was dragging us back, we were able to start looking for answers.
+
+A new architecture should provide an answer to each one of PrestaShop's main structural problems.
+
+#### Interaction delays
+
+The interactive web, since its dawn in the nineties, has been based on a process where the user opens an HTML page containing a form, fills it out and sends it to the server, who takes the form data, processes it, stores it in a database, and produces a new HTML and form which is sent back to the user, restarting the cycle. One of the problems of this pattern is that once the user performs an action, **they are blocked and must wait for the whole page to reload** before being able to interact with the page again.
+
+![Synchronous user interaction][user-interaction-sync]
+
+The _wait-for-reload-between-interactions_ problem has been basically fixed for a long time now. Modern architectures are usually based on front-end, Javascript-based applications that are loaded once and run in the browser. In this kinds of apps, all data exchange between the front end app and the server is performed asynchronously (ie. in a non-blocking manner) through XmlHttpRequests (also known as Ajax) or using Web Sockets. This brings enhanced responsiveness to the user interface, allowing the user to **continue interacting with the application while data exchange is performed**.
+
+![Asynchronous user interaction][user-interaction-async]
+
+Meanwhile, on the server side, information is served and received through some kind of API (Rest or GraphQL, for example) and in a language designed for data transport, like XML or JSON. Outputting data in these languages further enhances responsiveness thanks to a reduced volume of data travelling between between browser and server. In addition, by separating data from presentation, both the presentation and the data themselves can be processed more easily and efficiently. 
```

Comment:
I prefer to qualify XML and Json as format rather than language:
information is served and received through an API (Rest or GraphQL, for example) serving content structured with a format such as XML or Json

Reply:
Wdyt?

```suggestion
Meanwhile, on the server side, information is served and received through an API (Rest or GraphQL, for example) in a format designed for data transport, like XML or JSON. Outputting data in a structured format further enhances responsiveness, thanks to a reduced volume of data traveling between between browser and server. In addition, by separating data from presentation, both the presentation and the data themselves can be processed more easily and efficiently. 
```